### DISCUSSION TITLE
How to find row number?

### DISCUSSION THREAD

**Discussion by masnwilliams** [prisma/prisma#11746] [Q&A]
Hello! I have a waitlist table and want to show the user what number in the table they are.

I cannot find a way using Prisma to display the row number of their email.

I was thinking of getting the count of the waitlist, get all the waitlist rows, then finding the place of the user's email in that waitlist but there has to be an easier way right?

Thanks.

**Comment by nurul3101** [comment]
Hey @johnmasonwilliams 👋🏻 ,
How are you storing the waitlist number of each customer?
Are you inserting new customers sequentially in the waitlist table and removing them once they get the table?
Can you please share your schema file or your waitlist model?

**Reply by masnwilliams** [reply]
Right now the waitlist number is the id of the row as a pk.

Here is my schema for the Waitlist table:
```
model Waitlist {
  id        Int       @id @unique @default(autoincrement())
  email     String
  createdAt DateTime  @default(now())
  updatedAt DateTime? @updatedAt
}
```

I would just use the id but this is not accurate due to rows being removed from the table. So I want a way to find the actual row number a certain row is in a table.

**Reply by nurul3101** [reply]
You can use the [`row_number`](https://www.postgresql.org/docs/13/functions-window.html) function to get the actual row number of a particular row. (I am assuming that you are using PostgreSQL).
By combining [`queryRaw`](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#raw-queries-with-relational-databases) and `row_number` function, the actual row number can be found.

For example: 
```javascript
const result = await prisma.$queryRaw`select row_number() over(), * from waitlist`
```
will return the exact row numbers of each row.

Attached a reference image on how the response of select query would look like:

<img width="474" alt="Screenshot 2022-02-25 at 11 59 25 AM" src="https://user-images.githubusercontent.com/13049676/155665895-7dc41839-09dd-4146-8e8f-82037f853ffe.png">

Please let me know if this solves it for you?

**Reply by masnwilliams** [reply]
I'm actually using PlanetScale for my database which is MySQL. Would this query be different for mysql?

I would also adding a `where` to query an inputted email would look something like this?
```
const result = await prisma.$queryRaw`select row_number() over(), * from waitlist where email=${email}`
```

**Reply by ashemag** [reply]
finding that row_number in planetscale/mysql results in `Message: unknown error: syntax error at position`

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
